nabla/test/nablaCurve.ts

```diff
@@ -77,49 +77,58 @@ export default async function (environment: TestSuiteEnvironment) {
       assertApproxEqRel(
         await calcEffectiveDeposit(e(80, 18), e(100, 18), e(10, 18)),
         e(10.004777762470999, 18),
-        e(0.001, 18),
+        1000n,
         "#1 mismatch"
       );
     },
 
     async testDepositWithdrawal() {
-      const amm = await newVendingMachine(address(curve));
+      const amm = await newVendingMachine(address(curve), 18);
       await amm.init(e(950, 18), e(1000, 18));
       assertApproxEqRel(
         await amm.totalReserves(),
         e(950.0235738135982, 18),
-        e(0.001, 18),
+        1000n,
         "test_deposit_withdraw initial total reserves mismatch"
       );
       const deposited = await amm.deposit(e(1000, 18));
-      assertApproxEqRel(deposited, e(1000.012058440754, 18), e(0.01, 18), "test_deposit_withdraw lp mismatch");
+      assertApproxEqRel(deposited, e(1000.012058440754, 18), 1000n, "test_deposit_withdraw lp mismatch");
       assertApproxEqRel(
         await amm.totalReserves(),
         e(1950.0235738135982, 18),
-        e(0.001, 18),
+        1000n,
         "test_deposit_withdraw intermediate total reserves mismatch"
       );
       assertApproxEqRel(
         await amm.withdraw(deposited),
         e(1000, 18),
-        e(0.01, 18),
+        1000n,
         "test_deposit_withdraw withdrawn amount mismatch"
       );
       assertApproxEqRel(
         await amm.totalReserves(),
         e(950.0235738135982, 18),
-        e(0.001, 18),
+        1000n,
         "test_deposit_withdraw final total reserves mismatch"
       );
     },
 
     async testNoNegativeDeposits() {
       // Regression test: (0x58788cb94b1d7fc19, 0x5b6e16f2c648b620c, 0, 0x3e8)
       assertLt(
-        await curve.effectiveDeposit(e(102, 18), e(105.41165658966665, 18), 0, e(0.0000001, 18)),
+        await curve.inverseDiagonal(e(102, 18), e(105.41165658966665, 18), e(102, 18) + e(0.0000001, 18), 18),
         2n ** 255n - 1n,
         "Regression test #1 overflow"
       );
     },
+
+    async testDepositFuzz(deposit: bigint, deposit2: bigint) {
+      for (let decimals = 8n; decimals < 24n; decimals++) {
+        const amm = await newVendingMachine(address(curve), decimals);
+        await amm.init(2n * 10n ** decimals, 5n * 10n ** decimals);
+        await amm.deposit(deposit >> 176n);
+        await amm.swapInto(deposit2 >> 176n);
```

Comment:
Why are we right-shifting by 176 here?

Reply:
A little magic number here, I should have documented. 

Fuzz tests in forge allow to roughly specify the range of numbers by stating the integer format of the parameter. If you declare an argument as, say `uint128`, then forge will insert a random number between 0 and 2**128 - 1.

In Nabla we declare the test to use parameters of type `uint80` for the fuzz test (see [here](https://github.com/0xamberhq/curve/pull/4/files#diff-5ec5fc6591c384aee5c61b7328aa7ba41b611a258f627a3512e0fb8483835156R124)). 

Since this is not possible to do in TypeScript I considered it is the best to always assume random numbers between 0 and 2**256 - 1 and then scale them down as required, e.g., by bit shifting.

256 - 80 = 176.

Reply:
I added a comment to explain this.